core/src/main/java/io/kestra/core/models/tasks/logs/LogShipper.java

```diff
@@ -0,0 +1,13 @@
+package io.kestra.core.models.tasks.logs;
+
+import io.kestra.core.models.annotations.Plugin;
+import reactor.core.publisher.Flux;
+
+@Plugin
+public abstract class LogShipper implements io.kestra.core.models.Plugin {
+
+    protected String type;
+
+    public abstract Flux<LogRecord> sendLogs(Flux<LogRecord> logRecord);
```

Comment:
```suggestion
    public abstract void sendLogs(Flux<LogRecord> logRecord);
```

sending logs should not update the logs so I think it's been better to return void.
Keep in mind that plugins should not be trusted, they can come from custom JARs on user installation